gdal/apps/gdal_grid_lib.cpp

```diff
@@ -823,13 +833,45 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
     if ( psOptions->pszSQL )
         nBands++;
 
-    // FIXME
-    int nXSize = psOptions->nXSize;
-    if ( nXSize == 0 )
-        nXSize = 256;
-    int nYSize = psOptions->nYSize;
-    if ( nYSize == 0 )
-        nYSize = 256;
+    int nXSize;
+    int nYSize;
+    if ( psOptions->dfXRes && psOptions->dfYRes )
+    {
+        if (!(psOptions->dfXMax - psOptions->dfXMin) || !(psOptions->dfYMax - psOptions->dfYMin)) {
+            CPLError( CE_Failure, CPLE_IllegalArg,
+                    "Invalid txe or tye paramaters detected. Please check your -txe or -tye argument.");
+
+            if(pbUsageError)
+                *pbUsageError = TRUE;
+            return nullptr;
+        }
+
+        double dfXSize = (std::fabs(psOptions->dfXMax - psOptions->dfXMin) + (psOptions->dfXRes/2.0)) /
+            psOptions->dfXRes;
+        double dfYSize = (std::fabs(psOptions->dfYMax - psOptions->dfYMin) + (psOptions->dfYRes/2.0)) /
```

Comment:
I have one question. In the following code snippet, only one component uses fabs. Should it be done like that or is my way the correct way?

Code: https://github.com/OSGeo/gdal/blob/d88aabcb211233977ed202918182db23e12a3652/gdal/apps/gdalwarp_lib.cpp#L3233-L3238

Reply:
your way is fine (although xmax < xmin is a very special use case)